src/components/Filters/Filters.tsx

```diff
@@ -35,28 +36,42 @@ const FiltersContainer = styled(FlexContainer)`
 
 const Filters: React.FC<FiltersProps> = ({
   fields,
-  state = getDefaultState(fields),
+  state,
   onApply,
   onChange,
   onClose,
   onCancel,
   isLoading = false,
 }) => {
-  const [filtersValues, setFiltersValues] = useState<Array<Filter>>(state);
+  const [filtersValues, setFiltersValues] = useState<Array<Filter>>(undefined);
   const [isDefaultState, setIsDefaultState] = useState(true);
   const [hasUnappliedFilters, setHasUnappliedFilters] = useState(false);
 
   useEffect(() => {
```

Comment:
There are 3 useEffects here with shared dependencies. I understand we want to cover all the cases but is it possible to simplify?

Reply:
I understand you feel something suspicious 🙂 There are 3 useStates and each has own useEffect to set the state. If I would merge all useEffects into one with many if elses then some states would be updated even if not necessarily. I believe it is correct and I kind of like how it is separated, for me it is more clear.

Reply:
I've been considering if I shouldn't move it into custom hook

Reply:
My concern is that `fields` is a dependency of the first two useEffects, but I guess fields is a required prop and never changes. If that's correct, then it works :)